### PR TITLE
Feat: add add-ons Crossplane and AWS provider

### PR DESCRIPTION
**Pull request by zzxwill**
Added experimental add-ons for
- Crossplane
- AWS provider for Crossplane

Signed-off-by: Zheng Xi Zhou <zhengxi.zzx@alibaba-inc.com>

### Checklist

I have:

- [ ] Title of the PR starts with type (e.g. `[Addon]` or `[Trait]`).
- [ ] Updated/Added any relevant [documentation] and [examples].
- [ ] Unit/E2E Tests added.

### CODE REVIEW

**Review by oeular — COMMENTED** [review]
Pls change readme.md to README.md

**Review by oeular — COMMENTED** [review]
Maybe the crossplane.cue is needed to expose more parameters for users.

**Comment by zzxwill** [comment]
> Maybe the crossplane.cue is needed to expose more parameters for users.

Yeah, any suggestions for those parameters?

**Comment by wangyikewxgm** [comment]
Lets provide a example to tell user how to use this addon.

**Comment by wangyikewxgm** [comment]
Please add an icon.

**Comment by zzxwill** [comment]
@wangyikewxgm The icon for addon crossplane-aws and the example are added. Please help review it.

File: experimental/addons/crossplane-aws/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: crossplane-aws
+version: 0.0.1
+description: AWS Provider for Crossplane
+icon: https://static.kubevela.net/images/logos/aws.png
+url: https://crossplane.io/
+
+tags:
+  - Crossplane provider
```
Comment (wangyikewxgm) [inline experimental/addons/crossplane-aws/metadata.yaml:8]: “- cloud resource”？align with terraform.

Comment (zzxwill) [reply experimental/addons/crossplane-aws/metadata.yaml:8]: Yes, I did in addon `crossplane` https://github.com/kubevela/catalog/pull/376/files#diff-1f030f14d0e749a3b151ca6f5240a231928dd54545f64b5230f01ed4c5dcc387R8.

**Review by wangyikewxgm — DISMISSED** [review]

File: experimental/addons/crossplane/readme.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Crossplane
+
+Crossplane is an open source Kubernetes add-on that transforms your cluster into a universal control plane.
+Crossplane enables platform teams to assemble infrastructure from multiple vendors.
```
Comment (wonderflow) [inline experimental/addons/crossplane/readme.md:4]: let's add a reference to their website

**Review by wonderflow — DISMISSED** [review]

File: experimental/addons/crossplane/readme.md (hunk 2 above)
Comment (zzxwill) [reply experimental/addons/crossplane/readme.md:4]: Fixed. Also added the link for provider AWS.

**Review by wonderflow — APPROVED** [review]

**Pull request merged**